.gitignore

```diff
@@ -7,4 +7,9 @@
 .idea/
 .vs
 .vscode
-build
\ No newline at end of file
+build
+# generated on macOS
+*.cmake
+*.log
+CMakeFiles
+.DS_Store
```

Comment:
Umm... this stuff should be a build folder, which is included above. Maybe another commit?

Reply:
These files may be generated everywhere. Users may do something like this to store build log for example `./gen-linux.sh > gen-linux.log`. `.DS_Store` may be generated in every directories that are visited by Finder in macOS.

Reply:
Ehhh? What if they try to do `./gen-linx.sh > gen-linux.txt`? I wouldn't just exclude files willy nilly like that.

Reply:
I think `.log` is a widely used extension for storing `log`. I am not sure if that is the case for `.txt`. I am not saying that we should include every possible extensions that users might use - just acceptable ones.

Reply:
The *.log are personal preferences. I think unless you can't do anything about it (like *.cmake are generated by cmake tools, .DS_Store are generated by system, .obj by compiler), should not include it in .gitignore, just manage it yourself

Reply:
Ok. I will remove `*.log` entry.

Reply:
Wait wait, we already exclude CMakeFiles and `*.cmake` though by `build-*`, why do we have to exclude them again?

Reply:
They are generated in project directory when building on macOS. Just to make sure `cmake` configured information is stored in `CMakeList.txt`, isn't it?

Reply:
Ehh? I work on macos and I have no problem with any cmake files (other than the .DS_Store stuff, that's annoying). You should generate your cmake files in a subdirectory via `cmake ..` so you still have control over them. It's not a good idea to mix build files and source files anyway.

Reply:
I will re-check it. If it is you said, I will update the PR.

Reply:
Yeah. They aren't generated when I re-check the build. I updated the PR.